zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/launcher/SparkInterpreterLauncher.java

```diff
@@ -171,7 +171,7 @@ public SparkInterpreterLauncher(ZeppelinConfiguration zConf, RecoveryStorage rec
       sparkConfBuilder.append(" --proxy-user " + context.getUserName());
     }
 
-    env.put("ZEPPELIN_SPARK_CONF", escapeSpecialCharacter(sparkConfBuilder.toString()));
+    env.put("ZEPPELIN_SPARK_CONF", sparkConfBuilder.toString());
```

Comment:
Why remove escapeSpecialCharacter ? The purpose of escapeSpecialCharacter is for security purpose (to avoid command injection in `interpreter.sh`)

Reply:
Escaping 'ZEPPELIN_SPARK_CONF' is no longer necessary, because all values are passed to 'exec' as string arguments. The bash escaped string values automatically.
Because of `exec` the bash process is replaced by the jvm process, so it is not possible to insert further commands after `exec`.

What do you think?

Reply:
make sense. Thanks for explanation.